File: frontend/src/interfaces/cliente.ts
export type TipoCliente = 'PESSOA_FISICA' | 'PESSOA_JURIDICA';

export interface Cliente {
  id: string;
  nome: string;
  tipo: TipoCliente;
  documento: string;
  email?: string;
  telefone?: string;
  cnh?: string;
  cnhValidade?: string;
  
  cep?: string;
  logradouro?: string;
  numero?: string;
  complemento?: string;
  bairro?: string;
  cidade?: string;
  uf?: string;

  cnhUrl?: string;
  comprovanteResidenciaUrl?: string;
}

export interface ClienteRequest extends Omit<Cliente, 'id'> {}

File: models/cliente.py
from sqlalchemy import Column, String, DateTime, Text, Boolean
from core.database import Base
from datetime import datetime
from models.usuario import generate_uuid

class Cliente(Base):
    __tablename__ = "clientes"

    id = Column(String, primary_key=True, default=generate_uuid)
    tipo_pessoa = Column(String(20), nullable=False) # FISICA, JURIDICA
    nome = Column(String(150), nullable=False)
    documento = Column(String(20), unique=True, nullable=False, index=True) # CPF ou CNPJ
    cnh = Column(String(20), nullable=True)
    email = Column(String(100), nullable=True)
    telefone = Column(String(20), nullable=True)
    
    # Endereco
    cep = Column(String(10), nullable=True)
    logradouro = Column(String(150), nullable=True)
    numero = Column(String(20), nullable=True)
    complemento = Column(String(100), nullable=True)
    bairro = Column(String(100), nullable=True)
    cidade = Column(String(100), nullable=True)
    uf = Column(String(2), nullable=True)
    
    observacoes = Column(Text, nullable=True)
    ativo = Column(Boolean, default=True)
    criado_em = Column(DateTime, default=datetime.utcnow)
